File: leetcode/1423. Maximum Points You Can Obtain from Cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        l, r = 0, len(cardPoints) - k
        ans = sum(cardPoints[r:])
        max_points = ans
        while r < len(cardPoints):
            ans = ans + (cardPoints[l] - cardPoints[r])
            max_points = max(max_points, ans)
            l, r = l + 1, r + 1
        return max_points

File: leetcode/1423. Maximum Points You Can Obtain from Cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        int l = 0, r = n - k, sum = 0, result;
        for (int i = r ; i < n; i++) {
            sum += cardPoints[i];
        }
        result = sum;
        while (r < n) {
            sum += (cardPoints[l] - cardPoints[r]);
            result = max(result, sum);
            l++, r++;
        }
        return result;
    }
};